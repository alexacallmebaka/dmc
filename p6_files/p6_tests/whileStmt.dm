main : () int  {
  a : int = 10;
  b : bool = true;
  while (a > 0){
    a--;
  }
  a = 1;
  return a;
}
