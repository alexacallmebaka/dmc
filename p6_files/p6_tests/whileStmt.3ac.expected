[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN main LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            [a] := 10
            [b] := 1
lbl_1:      nop
            [tmp0] := [a] GT64 0
            IFZ [tmp0] GOTO lbl_2
            [a] := [a] SUB64 1
            goto lbl_1
lbl_2:      nop
            [a] := 1
            setret [a]
            goto lbl_0
lbl_0:      leave main
